src/netio.c

```diff
@@ -1452,6 +1452,7 @@ int pr_netio_read(pr_netio_stream_t *nstrm, char *buf, size_t buflen,
       }
 
       nstrm->strm_errno = 0;
+      errno = EOF;
```

Comment:
In the reproduction recipe, the tight loop _should_ have ended sooner, but the lack of properly setting this `errno` value prevented that, further allowing the denial-of-service to occur.  So we properly set `errno` to EOF, to catch another possible avenue for exploitation.